custom_components/hacs/api/hacs_repository_data.py

```diff
@@ -35,8 +36,13 @@ async def hacs_repository_data(hass, connection, msg):
         return
 
     if action == "add":
-        if "github." in repo_id:
-            repo_id = repo_id.split("github.com/")[1]
+        # Extract username/repo, stripping URL and trailing characters
+        pattern = r"(?:(?:.*github.com.)|^)([A-Za-z0-9-]+\/[A-Za-z0-9_.-]+)"
```

Comment:
@ludeeus can you see any important cases that this pattern would not match?

I have tested a few cases here - go to Unit Tests: https://regex101.com/r/VKkpVx/2

It will support all of the below, and extract the `user/repo` substring:

* `*github.com/user/repo/*`
* `user/repo/*`

Valid usernames consist of alphanumeric characters and dashes, while valid repo names consist of alphanumeric characters, underscores, dashes, and periods.

If a match isn't found (i.e. if nonsense is entered) then `repo_id` will not be modified, and it will 404 in the next step as it would have previously.

Reply:
Oh.. You actually did it :100: 

Was planing on looking into it later this week :joy: 

There is still one case that I have seen that still matches `https://github.com/my/repo.git`, the `.git` part should be striped as well